.github/workflows/ci-appimage.yml

```diff
@@ -48,7 +48,7 @@ jobs:
 
     - name: Upload logs (if failed)
       if: failure()
-      uses: actions/upload-artifact@master
+      uses: actions/upload-artifact@main
```

Comment:
Why no `actions/upload-artifact@v3` or `actions/upload-artifact@v4` ?

Reply:
`@main` is the approach [already used elsewhere](https://github.com/alire-project/alire/pull/1756/files#diff-b81b30f301824b3151eb4e7abf0802efe535b610860e830d1aad7f58bffe9862L92), and will avoid the need for further changes if there are future deprecations (which I assumed was the justification).

I have just changed them all to `@v4` (`@v3` will be deprecated in November), which I agree is the recommended approach.